src/DOOM/m_menu.c

```diff
@@ -1418,6 +1418,28 @@ void M_FinishReadThis(int choice)
     M_SetupNextMenu(&MainDef);
 }
 
+//
+// When a secret is found (ColleagueRiley)
+//
+void doom_secretFound(int s) {
+    static int lastTime = 0;
+    int usec;
+
+    if (lastTime == 0) {
+        doom_gettime(&lastTime, &usec);
+        S_StartSound(0, sfx_getpow);
+    }
+    
+    int curTime;
+    doom_gettime(&curTime, &usec);
```

Comment:
You should use the other gettime, that you used before you change it. Which counts in frames. That way it ensure the simulation is the same every time you replay or play a demo.

Also altho it went by really fast, it was fine I tested it. You can put it back to 20 frames. Especially if you found it removes player control if longer than that.

Reply:
I'm confused. Wouldn't it be more standardized if you go by seconds instead of frames? The time it takes for 20 seconds to pass remains the same but the time it takes for 20 frames to be drawn might differ.

Reply:
Also, there's most definitely a way to do this without freezing the player. I'll look into better ways of doing it.

Would you mind if the message is in the top left hand corner or would you prefer it in the middle, similar to gzdoom.

Reply:
No, you want a number of frames. Not real time. If the game sudently lags, then real time 20sec will be a different amount of time. So if I replay the exact same run, it won't be deterministic.

I prefer at the top left because doom already has the code for this, similar to how to pick up an item. So the code change will be smaller and less risky.